### PR TITLE
cloud: ovirt: add developer README.rst

### PR DESCRIPTION
**Pull request by machacekondra**
##### SUMMARY

Add developer README file.

##### ISSUE TYPE

 - Docs Pull Request

##### COMPONENT NAME

ovirt

##### ANSIBLE VERSION

```
2.3.0
```

##### ADDITIONAL INFORMATION

```

```

### CODE REVIEW

**Comment by machacekondra** [comment]
@mwperina

File: lib/ansible/modules/cloud/ovirt/README.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
+oVirt Ansible Modules
+=========================
+
+These are a set of modules for interacting with oVirt. This document serves
```
Comment (mykaul) [inline lib/ansible/modules/cloud/ovirt/README.md]: oVirt/RHV perhaps?

File: lib/ansible/modules/cloud/ovirt/README.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+as developer coding guidelines for modules.
+
+Naming
+------
+
+* All modules should start with `ovirt_`
```
Comment (mykaul) [inline lib/ansible/modules/cloud/ovirt/README.md]: start with -> an 'ovirt_' prefix

File: lib/ansible/modules/cloud/ovirt/README.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+* Every module should return the dictionary of the resource it manages.
+* Never change the name of the parameter, as we do guarantee backward compatibility,
+  instead if needed use aliases.
+* If some parameter can't achieve idempotency for any reason, please document it.
+
+Interoperability
+----------------
+
+* All modules should work appropriately against all minor versions of the
+  version 4 of the API. The version 3 of the API is not and don't have to
```
Comment (mykaul) [inline lib/ansible/modules/cloud/ovirt/README.md]: Rephrase: 'Version 3 of the API is not supported.'

File: lib/ansible/modules/cloud/ovirt/README.md [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+  with `ovirt_facts_full_argument_spec`.
+* Common functions to all modules should be implemeneted in `module_utils/ovirt.py` file,
+  so it can be reused.
+* Python SDK version 4 must be used.
+
+New module development
+----------------------
+
+Please first read this [link](http://docs.ansible.com/ansible/dev_guide/developing_modules.html#how-to-develop-a-module),
+to know what common things every module must have.
```
Comment (mykaul) [inline lib/ansible/modules/cloud/ovirt/README.md]: things -> properties, functions, features

File: lib/ansible/modules/cloud/ovirt/README.md [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+    # 'module.fail_json' in case of exception. Always use
+    # 'exception=traceback.format_exc' for debugging purposes:
+    module.fail_json(msg=str(e), exception=traceback.format_exc())
+finally:
+    # Logout only in case the user passed the 'token' in 'auth'
+    # parameter:
+    connection.close(logout=auth.get('token') is None)
+```
+
+Now, if your module must support some action handling, like for example
```
Comment (mykaul) [inline lib/ansible/modules/cloud/ovirt/README.md]: remove 'like'

**Review by mwperina — APPROVED** [review]
👍

**Comment by mattclay** [comment]
CI is now passing. There was a bug in the module validator which was causing CI to fail.

**Comment by mattclay** [comment]
@machacekondra New documentation should use reStructuredText instead of Markdown. We're slowly migrating away from using Markdown in most places.

**Comment by mattclay** [comment]
The validate-module errors are wrong. I'll fix the validator and re-run CI.

**Comment by mattclay** [comment]
The one remaining CI failure appears to be valid:

```
lib/ansible/modules/cloud/ovirt/README.rst:180:0: (python) expected an indented block
```

**Comment by machacekondra** [comment]
Thanks for help @mattclay

**Comment by machacekondra** [comment]
shipit

**Review by mwperina — APPROVED** [review]
👍

**Comment by machacekondra** [comment]
@ryansb

**Comment by ryansb** [comment]
Waiting on docs approval from @dharmabumstead

**Comment by dharmabumstead** [comment]
Review in progress. ETA ~Thursday 6/15 by 6pm PDT.

File: lib/ansible/modules/cloud/ovirt/README.rst [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+        # 'build_entity' method. You don't have to care about calling the update,
+        # it's called behind the scene by the 'BaseModule' class.
+        def update_check(self, entity):
+            return (
+                equal(self.param('comment'), entity.comment)
+                and equal(self.param('description'), entity.description)
+            )
+
+The previous part is the most important one, if you have it, you must
+create the basic flow of the module:
```
Comment (dharmabumstead) [inline lib/ansible/modules/cloud/ovirt/README.rst]: Please reword this; I'm unclear what you're trying to say here.

**Review by dharmabumstead — CHANGES_REQUESTED** [review]

**Comment by machacekondra** [comment]
Thank you for review, I've changed the requested paragraph.

**Comment by machacekondra** [comment]
@dharmabumstead please re-review.

**Comment by machacekondra** [comment]
@dharmabumstead Can you please re-review?

**Review by mwperina — APPROVED** [review]
Looks good to me, thanks! 👍

**Comment by mwperina** [comment]
@dharmabumstead Can you please re-review?

**Review by dharmabumstead — APPROVED** [review]
Docs are OK with a few minor edits. Still needs core team sign off before merging. Thanks @machacekondra!

**Pull request merged**